docs/services/gcp-services-guide.md

```diff
@@ -0,0 +1,536 @@
+
+Using the service account json `crossplane-gcp-provider-key.json`:
+* Generate Base64 encoded value to store in a `Secret`:
+```bash
+export BASE64ENCODED_GCP_PROVIDER_CREDS=$(base64 crossplane-gcp-provider-key.json | tr -d "\n")
+```
+
+* Define a GCP `Provider` and `Secret`:
+```bash
+cat > gcp-provider.yaml <<EOF
+---
+apiVersion: v1
+data:
+  credentials.json: $BASE64ENCODED_GCP_PROVIDER_CREDS
+kind: Secret
+metadata:
+  namespace: $INFRA_NAMESPACE
+  name: gcp-provider-creds
+type: Opaque
+---
+apiVersion: gcp.crossplane.io/v1alpha2
+kind: Provider
+metadata:
+  namespace: $INFRA_NAMESPACE
+  name: gcp-provider
+spec:
+  credentialsSecretRef:
+    name: gcp-provider-creds
+    key: credentials.json
+  projectID: $PROJECT_ID
+EOF
+
+kubectl apply -f gcp-provider.yaml
+
+```
+* Verify GCP provider was successfully registered by the crossplane
+```bash
+kubectl get providers.gcp.crossplane.io -n $INFRA_NAMESPACE
+kubectl get secrets -n $INFRA_NAMESPACE
+```
+
+#### Cloud-Specific Resource Classes
+Cloud-specific resource classes are used to define a reusable configuration for
+a specific managed service. Wordpress requires a MySQL database, which can be
+satisfied by a [Google Cloud SQL Instance](https://cloud.google.com/sql/docs/mysql/) instance.
+
+* Define a GCP CloudSQL class `CloudsqlInstanceClass`:
+```bash
+cat > gcp-mysql-standard.yaml <<EOF
+---
+apiVersion: database.gcp.crossplane.io/v1alpha2
+kind: CloudsqlInstanceClass
+metadata:
+  name: standard-cloudsql
+  namespace: $INFRA_NAMESPACE
+specTemplate:
+  databaseVersion: MYSQL_5_7
+  tier: db-n1-standard-1
+  region: us-central1
+  storageType: PD_SSD
+  storageGB: 10
+  # Note from GCP Docs: Your Cloud SQL instances are not created in your VPC network.
+  # They are created in the service producer network (a VPC network internal to Google) that is then connected (peered) to your VPC network.
+  privateNetwork: projects/$PROJECT_ID/global/networks/$NETWORK_NAME
+  providerRef:
+    name: gcp-provider
+    namespace: $INFRA_NAMESPACE
+  reclaimPolicy: Delete
+EOF
+
+kubectl apply -f gcp-mysql-standard.yaml
+```
+* You can verify creation with the following command and output:
+
+*Command*
+```bash
+kubectl get cloudsqlinstanceclass -n ${INFRA_NAMESPACE}
+```
+*Output*
+```bash
+NAME                   PROVIDER-REF   RECLAIM-POLICY   AGE
+standard-cloudsql      gcp-provider   Delete           11s
+```
+
+You are free to create more GCP `CloudsqlInstanceClass` instances to define more
+potential configurations. For instance, you may create `large-gcp-mysql` with
+field `storageGB: 100`.
+
+#### Namespaces
+Kubernetes namespaces allow for separation of environments within your cluster.
+You may choose to use namespaces to group resources by team, application, or any
+other logical distinction. For this demo, we will create a namespace called
+`app-project1-dev`, which we will use to group our Wordpress resources.
+
+* Create a `Namespace`:
```

Comment:
This line probably isn't needed; we could just do something more like this:

```
. . . which we will use to group our Wordpress resources:

\```
kubectl create . . . 
\```
```